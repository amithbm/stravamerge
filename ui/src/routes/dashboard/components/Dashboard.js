import React from 'react'
import { Grid, Row, Col } from 'react-bootstrap'
import ActivitySummary from '../../activities/components/ActivitySummary'
import ActivitiesListContainer from '../../activities/containers/ActivitiesListContainer'

const Dashboard = ({isLoading, pageSize, page, moveNext, movePrevious}) => isLoading ? <div></div> : <Grid>
    <Row>
        <Col xs={12} sm={7}>
            <ActivitiesListContainer pageSize={pageSize} page={page} moveNext={moveNext} movePrevious={movePrevious} />
        </Col>
        <Col xs={12} sm={5}>
            <ActivitySummary />
        </Col>
    </Row>
</Grid>

Dashboard.propType = {
    isLoading: React.PropTypes.bool.isRequired,
    pageSize: React.PropTypes.number.isRequired,
    page: React.PropTypes.number.isRequired,
    moveNext: React.PropTypes.func.isRequired,
    movePrevious: React.PropTypes.func.isRequired
}

export default Dashboard