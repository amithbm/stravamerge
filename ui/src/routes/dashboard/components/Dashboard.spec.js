import React from 'react'
import {shallow} from 'enzyme'
import Dashboard from './Dashboard'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<Dashboard />', () => {
        it('renders as expected when loaded', () => {
            const component =  shallow(<Dashboard isLoading={true} pageSize={10} page={0} page={[]} moveNext={() => 0} movePrevious={() => 0}  />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })

        it('renders as expected when not loaded', () => {
            const component =  shallow(<Dashboard isLoading={false} pageSize={10} page={0} page={[]} moveNext={() => 0} movePrevious={() => 0}  />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})

