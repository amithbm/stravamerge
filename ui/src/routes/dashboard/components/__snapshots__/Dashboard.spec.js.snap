exports[`components <Dashboard /> renders as expected when loaded 1`] = `<div />`;

exports[`components <Dashboard /> renders as expected when not loaded 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}>
  <Row
    bsClass="row"
    componentClass="div">
    <Col
      bsClass="col"
      componentClass="div"
      sm={7}
      xs={12}>
      <Connect(Component)
        moveNext={[Function]}
        movePrevious={[Function]}
        page={Array []}
        pageSize={10} />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      sm={5}
      xs={12}>
      <Component />
    </Col>
  </Row>
</Grid>
`;
